encoding/codecv3/codecv3.go

```diff
@@ -0,0 +1,249 @@
+	copy(result[:32], b.z[:])
+	copy(result[32:], y[:])
+
+	return result, nil
+}
+
+// BlobDataProofForPointEvaluation computes the abi-encoded blob verification data.
+func (b *DABatch) BlobDataProofForPointEvaluation() ([]byte, error) {
+	if b.blob == nil {
+		return nil, errors.New("called BlobDataProofForPointEvaluation with empty blob")
+	}
+	if b.z == nil {
+		return nil, errors.New("called BlobDataProofForPointEvaluation with empty z")
+	}
+
+	commitment, err := kzg4844.BlobToCommitment(b.blob)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create blob commitment")
+	}
+
+	proof, y, err := kzg4844.ComputeProof(b.blob, *b.z)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create KZG proof at point, err: %w, z: %v", err, hex.EncodeToString(b.z[:]))
+	}
+
+	// Memory layout of ``_blobDataProof``:
+	// | z       | y       | kzg_commitment | kzg_proof |
+	// |---------|---------|----------------|-----------|
+	// | bytes32 | bytes32 | bytes48        | bytes48   |
+
+	values := []interface{}{*b.z, y, commitment, proof}
+	blobDataProofArgs, err := codecv1.GetBlobDataProofArgs()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get blob data proof args, err: %w", err)
+	}
+	return blobDataProofArgs.Pack(values...)
+}
+
+// Blob returns the blob of the batch.
+func (b *DABatch) Blob() *kzg4844.Blob {
+	return b.blob
+}
+
+// EstimateChunkL1CommitBatchSizeAndBlobSize estimates the L1 commit uncompressed batch size and compressed blob size for a single chunk.
+func EstimateChunkL1CommitBatchSizeAndBlobSize(c *encoding.Chunk) (uint64, uint64, error) {
+	return codecv2.EstimateChunkL1CommitBatchSizeAndBlobSize(c)
+}
+
+// EstimateBatchL1CommitBatchSizeAndBlobSize estimates the L1 commit uncompressed batch size and compressed blob size for a batch.
+func EstimateBatchL1CommitBatchSizeAndBlobSize(b *encoding.Batch) (uint64, uint64, error) {
+	return codecv2.EstimateBatchL1CommitBatchSizeAndBlobSize(b)
+}
+
+// EstimateChunkL1CommitCalldataSize calculates the calldata size needed for committing a chunk to L1 approximately.
+func EstimateChunkL1CommitCalldataSize(c *encoding.Chunk) uint64 {
+	return codecv2.EstimateChunkL1CommitCalldataSize(c)
+}
+
+// EstimateBatchL1CommitCalldataSize calculates the calldata size in l1 commit for this batch approximately.
+func EstimateBatchL1CommitCalldataSize(b *encoding.Batch) uint64 {
+	return codecv2.EstimateBatchL1CommitCalldataSize(b)
+}
+
+// EstimateBlockL1CommitGas calculates the total L1 commit gas for this block approximately.
+func EstimateBlockL1CommitGas(b *encoding.Block) uint64 {
+	return codecv2.EstimateBlockL1CommitGas(b)
+}
+
+// EstimateChunkL1CommitGas calculates the total L1 commit gas for this chunk approximately.
+func EstimateChunkL1CommitGas(c *encoding.Chunk) uint64 {
+	return codecv2.EstimateChunkL1CommitGas(c)
+}
+
+// EstimateBatchL1CommitGas calculates the total L1 commit gas for this batch approximately.
+func EstimateBatchL1CommitGas(b *encoding.Batch) uint64 {
```

Comment:
This need to change to account for the point-evaluation precompile call.

Reply:
nice catch, added in [4de8a88](https://github.com/scroll-tech/da-codec/pull/11/commits/4de8a88c6a939e5c0c8735346423e7ce4d2a2142).